### PR TITLE
Enable scribe environment for compile-time benchmarks if requested.

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #133891

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @rec @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @xmfan

### CODE REVIEW

File: .github/workflows/pull.yml [hunk 1]
```diff
@@ -570,6 +607,7 @@ jobs:
       docker-image: ${{ needs.linux-focal-py3_12-clang10-experimental-split-build.outputs.docker-image }}
       test-matrix: ${{ needs.linux-focal-py3_12-clang10-experimental-split-build.outputs.test-matrix }}
       timeout-minutes: 600
+    secrets: inherit
```
Comment (malfet) [inline .github/workflows/pull.yml:610]: Not sure if `secrets: inherit` is better than passing explicit secrets workflow might need

**Review by malfet — APPROVED** [review]

File: .github/workflows/pull.yml (hunk 1 above)
Comment (ezyang) [reply .github/workflows/pull.yml:610]: For this to be profitable, there would have to be a case where one instantiation of linux_build.yml needs secrets, but another does not. For the one secret I have in scribe-pr right now, there is no reason to hide it, it should apply uniformly everywhere. So inherit seems like the right tool for the job.

File: benchmarks/dynamo/pr_time_benchmarks/benchmark_base.py [hunk 2]
```diff
@@ -58,7 +104,10 @@ def print(self):
     def collect_all(self):
         self.results = []
         if self._instruction_count:
-            self.results.append(
-                (self.name(), "instruction_count", self.count_instructions())
+            r = self.count_instructions()
+            self.results.append((self.name(), "instruction_count", r))
+            scribe_log_torch_benchmark_compile_time(
```
Comment (laithsakka) [inline benchmarks/dynamo/pr_time_benchmarks/benchmark_base.py:109]: I am unsure if we shall log here. When we run A/B this will just send both to scribe.

**Pull request closed without merging**